homeassistant/components/unifi/sensor.py

```diff
@@ -192,6 +199,85 @@ def async_device_state_value_fn(hub: UnifiHub, device: Device) -> str:
     return DEVICE_STATES[device.state]
 
 
+@callback
+def async_client_wan_monitor_supported_fn(
+    monitor_target: str, hub: UnifiHub, obj_id: str
+) -> bool:
+    """Determine if a client supports WAN monitoring."""
+    device = hub.api.devices[obj_id]
+
+    uptime_stats = device.uptime_stats
+    if uptime_stats is None:
+        return False
+
+    wan = uptime_stats["WAN"]
+
+    monitors = wan["monitors"]
+
+    return any(monitor_target in item["target"] for item in monitors)
+
+
+@callback
+def async_client_wan_monitor_latency(
+    wan: Literal["WAN", "WAN2"],
+    monitor_target: str,
+    hub: UnifiHub,
+    device: Device | None,
+) -> int | None:
+    """Retrieve the monitor target from WAN monitors."""
+
+    if device is None or device.uptime_stats is None:
+        return None
+
+    uptime_stats = device.uptime_stats
+
+    if uptime_stats_wan := uptime_stats.get(wan):
+        for item in uptime_stats_wan["monitors"]:
+            if monitor_target in item["target"]:
+                return item.get("latency_average", 0)
+
+    return None
+
+
+@callback
+def make_wan_latency_entity_description(
+    wan: Literal["WAN", "WAN2"], name: str, monitor_target: str
+) -> UnifiSensorEntityDescription:
+    """Create a WAN latency sensor entity description."""
+
+    return UnifiSensorEntityDescription[Devices, Device](
+        key=f"{name} WAN latency",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        entity_registry_enabled_default=False,
+        api_handler_fn=lambda api: api.devices,
+        available_fn=async_device_available_fn,
+        device_info_fn=async_device_device_info_fn,
+        name_fn=lambda _: f"{name} WAN latency",
+        object_fn=lambda api, obj_id: api.devices[obj_id],
+        supported_fn=lambda hub, obj_id: async_client_wan_monitor_latency(
+            wan, monitor_target, hub, hub.api.devices[obj_id]
+        )
+        is not None,
```

Comment:
This won't work, you should do something like, as the supported function is expected to only get `Callable[[UnifiHub, str], bool]`
```suggestion
        supported_fn=partial(async_client_wan_monitor_latency, wan, monitor_target),
```
This is probably the reason the coverage fails on async_client_wan_monitor_supported_fn

Reply:
Not sure `partial(async_client_wan_monitor_latency, wan, monitor_target)` will work, as the signature of `value_fn` and `supported_fn` are different. 

I at least get this locally `Argument "supported_fn" to "UnifiSensorEntityDescription" has incompatible type "partial[int | None]"; expected "Callable[[UnifiHub, str], bool]"  [arg-type]`

I think it's better to add the supported function again.

Reply:
Hmm, but you didn't use the supported_fn, weren't you supposed to change that? :)

Reply:
Based on feedback it was suggested that I should reuse the value method and use a lambda. 
But I can revert, and re-add the support_fn that then uses the value fn to avoid duplicated code.

Reply:
Sorry. I agree without understanding the implications then